src/core/include/openvino/core/dimension.hpp

```diff
@@ -177,6 +181,9 @@ class OPENVINO_API Dimension {
         swap(a.m_table_of_equivalence, b.m_table_of_equivalence);
     }
 
+    /// \brief String representation of Dimension
+    std::string to_string() const;
```

Comment:
Why do we need this method? We already have operator << for the same purpose.

Reply:
As I understood, it was needed as alignment with Layout, which had a "to_string()" @slyalin do we need both methods or << is enough?

Reply:
`to_string` is more convenient to be used in pure string expressions. @ilyachur are you ready to use `std::ostringstream` every time when you need to form some diagnostics output? I think you can decide based on usages we have.